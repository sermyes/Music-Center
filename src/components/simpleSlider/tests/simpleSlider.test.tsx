import SimpleSlider from '../simpleSlider';
import '@testing-library/jest-dom';
import renderer from 'react-test-renderer';
import { render, screen } from '@testing-library/react';
import userEvent from '@testing-library/user-event';

interface VideoItemProps {
  video: fakeVideo;
}
type fakeVideo = {
  id: number;
};
const VideoItem = ({ video }: VideoItemProps) => {
  return (
    <div>
      <p>{video.id}</p>
    </div>
  );
};

describe('SimpleSlider', () => {
  const settings = {
    onArrow: true,
    onDots: true,
    slideToShow: 3
  };

  const videos = [
    { id: 1 },
    { id: 2 },
    { id: 3 },
    { id: 4 },
    { id: 5 },
    { id: 6 }
  ];

  it('renders correctly', () => {
    const component = renderer.create(
      <SimpleSlider settings={settings}>
        {videos.map((video) => (
          <VideoItem key={video.id} video={video} />
        ))}
      </SimpleSlider>
    );
    expect(component.toJSON()).toMatchSnapshot();
  });

  it('renders Slier with arrow and dots', async () => {
    render(
      <SimpleSlider settings={settings}>
        {videos.map((video) => (
          <VideoItem key={video.id} video={video} />
        ))}
      </SimpleSlider>
    );

    const dots = screen.getAllByRole('listitem');
    expect(dots.length).toBe(2);
    const arrows = screen.getAllByTestId('arrow');
    expect(arrows.length).toBe(2);

    await userEvent.click(arrows[0]);
    expect(dots[1]).toHaveClass('active');

    await userEvent.click(arrows[1]);
    expect(dots[0]).toHaveClass('active');

    await userEvent.click(dots[1]);
    expect(dots[1]).toHaveClass('active');
  });

  it('renders Slier without arrow and dots', async () => {
    const settingsWithoutOption = {
      onArrow: false,
      onDots: false,
      slideToShow: 3
    };
    render(
      <SimpleSlider settings={settingsWithoutOption}>
        {videos.map((video) => (
          <VideoItem key={video.id} video={video} />
        ))}
      </SimpleSlider>
    );

    const dots = screen.queryAllByRole('listitem');
    expect(dots).toStrictEqual([]);
    const arrows = screen.queryAllByTestId('arrow');
    expect(arrows).toStrictEqual([]);
  });
});
