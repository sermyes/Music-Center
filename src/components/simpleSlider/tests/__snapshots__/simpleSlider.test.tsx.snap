// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SimpleSlider renders correctly 1`] = `
<div
  className="slider simple__slider"
>
  <div
    className="slidesContainer"
    data-testid="container"
  >
    <div
      className="slides"
      data-testid="slides"
    >
      <div
        className="slide"
      >
        <div>
          <p>
            1
          </p>
        </div>
      </div>
      <div
        className="slide"
      >
        <div>
          <p>
            2
          </p>
        </div>
      </div>
      <div
        className="slide"
      >
        <div>
          <p>
            3
          </p>
        </div>
      </div>
      <div
        className="slide"
      >
        <div>
          <p>
            4
          </p>
        </div>
      </div>
      <div
        className="slide"
      >
        <div>
          <p>
            5
          </p>
        </div>
      </div>
      <div
        className="slide"
      >
        <div>
          <p>
            6
          </p>
        </div>
      </div>
    </div>
  </div>
  <div
    className="arrow"
  >
    <button
      className="prev"
      data-testid="arrow"
      onClick={[Function]}
    >
      <i
        className="icon fas fa-chevron-left"
      />
    </button>
    <button
      className="next"
      data-testid="arrow"
      onClick={[Function]}
    >
      <i
        className="icon fas fa-chevron-right"
      />
    </button>
  </div>
  <ul
    className="dots simple__dots"
    onClick={[Function]}
  />
</div>
`;
